src/bluetooth/qbluetoothtransferreply_bluez.cpp

```diff
@@ -91,12 +82,63 @@ QBluetoothTransferReplyBluez::QBluetoothTransferReplyBluez(QIODevice *input, con
 */
 QBluetoothTransferReplyBluez::~QBluetoothTransferReplyBluez()
 {
-    QDBusConnection::sessionBus().unregisterObject(m_agent_path);
+    delete objectTransfer;
+    delete objectPush;
     delete client;
 }
 
+void QBluetoothTransferReplyBluez::fail(QBluetoothTransferReply::TransferError err,
+	    const QString msg)
+{
+    m_finished = true;
+    m_running = false;
+    m_errorStr = msg;
+    m_error = err;
+    QMetaObject::invokeMethod(this, "finished",
+			      Qt::QueuedConnection,
+			      Q_ARG(QBluetoothTransferReply*, this));
+}
+
 bool QBluetoothTransferReplyBluez::start()
 {
+    QString dest = request().address().toString();
+    QVariantMap args;
+    args.insert(QString::fromLatin1("Target"), QString::fromLatin1("OPP"));
```

Comment:
Use QStringLiteral() instead of QString::fromLatin1() so that the string constant is store in rodata section and there is no run-time overhead.